faq/faq.rst

```diff
@@ -105,6 +105,23 @@ Why does pg_relation_size report zero bytes for a distributed table?
 
 The data in distributed tables lives on the worker nodes (in shards), not on the coordinator. A true measure of distributed table size is obtained as a sum of shard sizes. Citus provides helper functions to query this information. See :ref:`table_size` to learn more.
 
+Why am I seeing an error about max_intermediate_result_size?
+------------------------------------------------------------
+
+Citus has to use more than one step to run some queries having subqueries or CTEs. It copies subquery results into temporary :ref:`reference_tables` for use by the main query. If these results are too large, this might cause unacceptable network overhead. Citus has a configurable setting, ``citus.max_intermediate_result_size`` to specify a subquery result size threshold at which the query will be canceled. If you run into the error, it looks like:
```

Comment:
Again the note that the data isn't stored in reference tables. Remove the link/text here.
Should we link to the section which talks about the push-pull architecture? This can help expand on their understanding.

Reply:
> If these results are too large, this might cause unacceptable network overhead.

We should note that the results are actually pulled to the coordinator. So apart from network overhead, an unbounded data size could cause the disk to fill up on the master, causing crashes and availability outages.